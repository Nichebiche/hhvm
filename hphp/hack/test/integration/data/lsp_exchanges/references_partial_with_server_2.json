[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {
        "delayUntilDoneInit": true
      },
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$test/waitForHhServerReady",
    "params": {}
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 50,
        "character": 27
      },
      "context": {
        "includeDeclaration": true
      },
      "partialResultToken": "prt123"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$test/waitForResponse",
    "params": {
      "id": 12
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 14,
    "method": "textDocument/references",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 50,
        "character": 27
      },
      "context": {
        "includeDeclaration": true
      },
      "partialResultToken": "partialResultToken2"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$test/waitForResponse",
    "params": {
      "id": 14
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  },
  {
    "jsonrpc": "2.0",
    "method": "exit",
    "params": {}
  }
]
