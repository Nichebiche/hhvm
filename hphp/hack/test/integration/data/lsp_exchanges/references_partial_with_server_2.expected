[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": true,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\"",
            "{",
            "#"
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": {
          "resolveProvider": true
        },
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "implementationProvider": true,
        "rageProvider": true,
        "experimental": {
          "snippetTextEdit": true,
          "autoCloseJsx": true
        }
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "result": []
  },
  {
    "jsonrpc": "2.0",
    "method": "$/progress",
    "params": {
      "token": "prt123",
      "value": [
        {
          "uri": "file://${root_path}/references.php",
          "range": {
            "start": {
              "line": 50,
              "character": 25
            },
            "end": {
              "line": 50,
              "character": 28
            }
          }
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$/progress",
    "params": {
      "token": "prt123",
      "value": [
        {
          "uri": "file://${root_path}/references2.php",
          "range": {
            "start": {
              "line": 4,
              "character": 15
            },
            "end": {
              "line": 4,
              "character": 18
            }
          }
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 14,
    "result": []
  },
  {
    "jsonrpc": "2.0",
    "method": "$/progress",
    "params": {
      "token": "partialResultToken2",
      "value": [
        {
          "uri": "file://${root_path}/references.php",
          "range": {
            "start": {
              "line": 50,
              "character": 25
            },
            "end": {
              "line": 50,
              "character": 28
            }
          }
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$/progress",
    "params": {
      "token": "partialResultToken2",
      "value": [
        {
          "uri": "file://${root_path}/references2.php",
          "range": {
            "start": {
              "line": 4,
              "character": 15
            },
            "end": {
              "line": 4,
              "character": 18
            }
          }
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]
